### DISCUSSION TITLE
Failed RPC on Linea network

### DISCUSSION THREAD

**Discussion by williamliangwl** [WalletConnect/walletconnect-monorepo#4393] [Developer Support]
When trying to buy Linea NFT on OKX, there's failed RPC when the wallet is connected via WalletConnect

https://rpc.walletconnect.com/v1/?chainId=eip155:59144&projectId={projectId}

Response:
```
{
    "status": "FAILED",
    "reasons": [
        {
            "field": "chainId",
            "description": "We don't support the chainId you provided: eip155:59144. See the list of supported chains here: https://docs.walletconnect.com/cloud/blockchain-api#supported-chains"
        }
    ]
}
```

The failure RPC also happens when connecting Metamask through WalletConnect to the OKX

Steps to reproduce:
1. Connect Metamask through WalletConnect to OKX
2. Find a Linea NFT
3. Buy NFT
4. It will ask to switch network, approve it
5. RPC fails (can be checked through network tab in browser inspector)

**Comment by glitch-txs** [comment]
You'll need to use a custom RPC provider when interacting with Linea

**Reply by williamliangwl** [reply]
sorry for noob question, the custom RPC Provider is added on the dApp or the wallet?

**Reply by glitch-txs** [reply]
Oh wait, do you mean OKX website? Or which dapp? This should be implemented in the dapp side.

**Reply by williamliangwl** [reply]
yes OKX website here https://www.okx.com/web3

**Reply by glitch-txs** [reply]
we're working on a fix! Thanks for reporting 🙏

**Reply by williamliangwl** [reply]
thanks for helping! let me know if the fix has been landed or if there's anything I can help with

**Reply by williamliangwl** [reply]
sorry @glitch-txs , can I get rough estimate / update on this fix? 🙏

**Reply by glitch-txs** [reply]
Should be supported now

**Reply by williamliangwl** [reply]
it looks like working now, thank you! @glitch-txs
